music_rnn/src/model.ts

```diff
@@ -27,10 +27,11 @@ const CELL_FORMAT = 'multi_rnn_cell/cell_%d/basic_lstm_cell/';
  *
  * A MusicRNN is an LSTM-based language model for musical notes.
  */
-export class MusicRNN {
+export class MusicRNN<T extends magenta.controls.ControlSignalUserArgs> {
```

Comment:
Does T have to be defined when the MusicRNN is created? If so, does that mean we have to change how we create MusicRNNs from a checkpoint where the T is defined in the checkpoint?

Reply:
T is inferred based on the ControlSignal passed to the MusicRNN constructor.  I think as long as the JSON is a valid ControlSignalSpec everything should check out.  There's a similar step that happens in the initialize() method of MusicRNN, where the JSON response is assumed to be a ConverterSpec.